tests/pytests/unit/modules/test_zypperpkg.py

```diff
@@ -254,3 +268,263 @@ def zypper_search_mock(name, *_args, **_kwargs):
         ret = zypper.list_holds()
         assert len(ret) == 1
         assert "bar-2:2.3.4-2.1.*" in ret
+
+
+@pytest.mark.parametrize(
+    "zypper_version,lowpkg_version_cmp,expected_inst_avc,expected_dup_avc",
+    [
+        ("0.5", [-1, -1], False, False),
+        ("1.11.34", [0, -1], False, True),
+        ("1.14.8", [0, 0], True, True),
+    ],
+)
+def test_refresh_zypper_flags(
+    zypper_version, lowpkg_version_cmp, expected_inst_avc, expected_dup_avc
+):
+    with patch(
+        "salt.modules.zypperpkg.version", MagicMock(return_value=zypper_version)
+    ), patch.dict(
+        zypper.__salt__,
+        {"lowpkg.version_cmp": MagicMock(side_effect=lowpkg_version_cmp)},
+    ):
+        _zypper = zypper._Zypper()
+        _zypper.refresh_zypper_flags()
+        assert _zypper.inst_avc == expected_inst_avc
+        assert _zypper.dup_avc == expected_dup_avc
+
+
+@pytest.mark.parametrize(
+    "inst_avc,dup_avc,avc,allowvendorchange_param,novendorchange_param,expected",
+    [
+        # inst_avc = True, dup_avc = True
+        (True, True, False, False, False, True),
+        (True, True, False, True, False, True),
+        (True, True, False, False, True, False),
+        (True, True, False, True, True, True),
+        # inst_avc = False, dup_avc = True
+        (False, True, False, False, False, True),
+        (False, True, False, True, False, True),
+        (False, True, False, False, True, False),
+        (False, True, False, True, True, True),
+        # inst_avc = False, dup_avc = False
+        (False, False, False, False, False, False),
+        (False, False, False, True, False, False),
+        (False, False, False, False, True, False),
+        (False, False, False, True, True, False),
+    ],
+)
+@patch("salt.modules.zypperpkg._Zypper.refresh_zypper_flags", MagicMock())
+def test_allow_vendor_change(
+    inst_avc,
+    dup_avc,
+    avc,
+    allowvendorchange_param,
+    novendorchange_param,
+    expected,
+):
+    _zypper = zypper._Zypper()
+    _zypper.inst_avc = inst_avc
+    _zypper.dup_avc = dup_avc
+    _zypper.avc = avc
+    _zypper.allow_vendor_change(allowvendorchange_param, novendorchange_param)
+    assert _zypper.avc == expected
```

Comment:
Do we need to upstream those tests? I don't see them in the upstream PR.

Reply:
We need to push those to either https://github.com/saltstack/salt/pull/60421 or https://github.com/saltstack/salt/pull/62033, depends on which will be merged first.